ansible/roles/jenkins-worker/meta/main.yml

```diff
@@ -5,4 +5,5 @@
 #
 
 dependencies:
+- { 'role': 'java-base' }
```

Comment:
removed in #1723, just for debugging or speedup during dev hopefully

Reply:
It was removed because the dependencies were wrong.  The task that depended on it needed to create some directory locations before it would be able to be installed.

Reply:
Instead, it was integrated into the task itself in the right place to satisfy ordering.

Reply:
So this is broken for the smartos install at least, it doesn't get java installed without it. How are we supposed to handle this now?

Reply:
https://github.com/nodejs/build/blob/master/ansible/roles/jenkins-worker/tasks/main.yml#L298-L299

Reply:
^--- that stanza should be hit, does it not get triggered on smartos?

Reply:
ah, here's the problem: that java-base inclusion is probably OK but it's too late because it comes after the service call `service: name=jenkins state=restarted enabled=yes` - so that's going to try and start jenkins but if there's no Java it'll fail, which is exactly where I was failing.

In my latest commit here, 436b080, I've removed it again from meta/main.yml and instead have moved up the `include_role` to just before the first startup (`service`) call. I don't have a bare host to test on but re-ran it on one of the smartos18 machines and watched the output and it appeared to do it in a workable order.

This isn't ideal IMO because an `include_role` in the middle of a set of tasks is going to get confusing for future debugging. It's probably technical debt we have to accept for now but some more refactoring is probably in order for jenkins-worker.

PTAL at latest.

Reply:
Your explanation makes sense to me, I ran into the issue on a fresh install, so didn't hit the restart problem.

Reply:
@rvagg thanks for figuring that out. My bad as I made the move in Sam's PR.  Not sure why we did not see the failure when running ansible on the machine I had to make the change for but glad you have a single solution at least for now.